modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPDefinitionLocalServiceTest.java

```diff
@@ -634,6 +638,64 @@ public void testUpdateCPDefinitionExternalReferenceCode() throws Exception {
 		Assert.assertEquals("ERC", cProduct.getExternalReferenceCode());
 	}
 
+	@Test
+	public void testUpdateCPDefinitionWithVersioningEnabled() throws Exception {
+		frutillaRule.scenario(
+			"Update product definition with versioning enabled"
+		).given(
+			"I add a product definition"
+		).when(
+			"the product versioning is enabled"
+		).and(
+			"the product is updated"
+		).then(
+			"the product should have a new version with the product change"
+		);
+
+		CPDefinition cpDefinition = CPTestUtil.addCPDefinitionFromCatalog(
+			_commerceCatalog.getGroupId(), SimpleCPTypeConstants.NAME, false,
+			false);
+
+		long cpDefinitionId = cpDefinition.getCPDefinitionId();
```

Comment:
I think this can be just inlined (also because I don't get why this is outside the try)

Reply:
what about also doing the same thing (update + assert before the try) to verify that without the setting the update doesn't create the version while in the try it creates it?